### DISCUSSION TITLE
Public projects using KubeOps

### DISCUSSION THREAD

**Discussion by lairdm** [dotnet/dotnet-operator-sdk#331] [Q&A] [answered]
Hi, this looks like a fantastic library, thank you very much for starting this project. I'm still also in the process of learning Kubernetes Operators in general, I just stumbled upon the concept reading an article last night, so I apologize if this is a silly question.

Are there any repos of public projects using KubeOps that I (or future people reading this) could examine? I've learnt over my career I'm personally better learning by example rather than API docs, so if anyone has example projects using this library I'd be very appreciative.

Thanks again for building this!

**Comment by buehler** [comment, marked as answer]
Hey @lairdm 

Yes, I coded the "hosted database operator" for my company a while ago: https://github.com/smartive/hosted-database-operator
Furthermore, I'm currently writing an operator for my master's degree which will be open sourced as soon as it's finished (in the next two months)

**Reply by lairdm** [reply]
Fantastic, thank you. I'm still wrapping my head around operators and this library, but thank you again for taking the time to develop it.

**Reply by buehler** [reply]
You're welcome :-)

The documentation (https://buehler.github.io/dotnet-operator-sdk/) should give insights on how to use the SDK while the Kubernetes website explains how operators work. Additionally, for my degree I had do describe them as well.

Resources:
- https://kubernetes.io/docs/concepts/extend-kubernetes/operator/
- https://buehler.github.io/mse-project-thesis-1/report.pdf (pages 11-13)
- and an "exercise" about operators and how to write one: https://buehler.github.io/mse-project-thesis-1/report.pdf (Appendix C, pages 62 ff.)
